pkg/build/api/v1beta3/types.go

```diff
@@ -382,6 +386,21 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// Shell is an action that executes a shell script in a container
+	// running the build output image.
+	Shell string `json:"shell,omitempty" description:"shell script to be executed in a container running the build output image"`
```

Comment:
I think I'd accept this pattern if it was:

```
ShellArgs []string
Args []string
Command []string
```

Specifying ShellArgs and either Args or Command is a validation error.  Specifying Args and Command together is valid.  ShellArgs must be an array - otherwise there is no way for callers to guarantee that that proper quoting is preserved across all runtimes.  The implementation of ShellArgs should follow the same patterns as described before.  Taking just a string loses context for the program and makes escaping harder for end users (bash escaping is hard to get right, we should help users do it).

I would approve an API change with those tweaks.